alf/nest/nest.py

```diff
@@ -539,8 +542,18 @@ def fast_map_structure(func, *structure):
     return pack_sequence_as(structure[0], [func(*x) for x in entries])
 
 
-def py_pack_sequence_as(nest, flat_seq):
-    """Returns a given flattened sequence packed into a given structure."""
+def py_pack_sequence_as(nest, flat_seq, keep_fields_order=False):
+    """Returns a given flattened sequence packed into a given structure.
+
+    Args:
+        nest: a nested structure
+        flat_seq: a flattened sequence to be packed into the same structure as
+            ``nest``
+        keep_fields_order: if true, do not sort the fields when packing for
```

Comment:
Shouldn't we have `keep_fields_order` for flatten() too in order to restore the original nest?

Reply:
> Shouldn't we have `keep_fields_order` for flatten() too in order to restore the original nest?

Added; for now only py_flatten supports this order arg.